### PR TITLE
Added clients implementation

### PR DESCRIPTION
**Pull request by umbreak**

### CODE REVIEW

File: cli/shared/src/test/scala/ch/epfl/bluebrain/nexus/cli/SparqlClientSpec.scala [hunk 1]
```diff
@@ -1,6 +1,85 @@
...
 
-class SparqlClientSpec extends AnyWordSpecLike with Matchers {}
+class SparqlClientSpec extends AnyWordSpecLike with Matchers with Fixtures with OptionValues {
+
+  private val organization  = Label("myorg")
+  private val project       = Label("mylabel")
+  private val query         = "SELECT * {?s ?p ?o} LIMIT 10"
+  private val randomViewUri = nxv / genString()
+
+  private val mockedHttpApp = HttpApp[IO] {
```
Comment (wwajerowicz) [inline cli/shared/src/test/scala/ch/epfl/bluebrain/nexus/cli/SparqlClientSpec.scala]: maybe using `mockito` would be a bit cleaner here?

File: cli/shared/src/main/scala/ch/epfl/bluebrain/nexus/cli/types/EventEnvelope.scala [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package ch.epfl.bluebrain.nexus.cli.types
 
 /**
- * Event wrapper adding the offset metadata.
+  * Event wrapper adding the offset metadata.
  **/
-final case class EventEnvelope(offset: Offset, event: Event)
+final case class EventEnvelope(offsetOpt: Option[Offset], event: Event)
```
Comment (wwajerowicz) [inline cli/shared/src/main/scala/ch/epfl/bluebrain/nexus/cli/types/EventEnvelope.scala]: is there a case where an event wouldn't have an offset?

File: cli/shared/src/test/scala/ch/epfl/bluebrain/nexus/cli/SparqlClientSpec.scala (hunk 1 above)
Comment (umbreak) [reply cli/shared/src/test/scala/ch/epfl/bluebrain/nexus/cli/SparqlClientSpec.scala]: Yes, maybe. I tried to do it without mockito from what we discussed lately. But this case is simple enough to be able to do it in both ways.

Let's see what @bogdanromanx thinks.

File: cli/shared/src/main/scala/ch/epfl/bluebrain/nexus/cli/types/EventEnvelope.scala (hunk 2 above)
Comment (umbreak) [reply cli/shared/src/main/scala/ch/epfl/bluebrain/nexus/cli/types/EventEnvelope.scala]: In our case probably not, but the standard, the API and the `ServerSent` model of `fs2` all allow for this field to be optional.

File: cli/shared/src/test/scala/ch/epfl/bluebrain/nexus/cli/SparqlClientSpec.scala (hunk 1 above)
Comment (bogdanromanx) [reply cli/shared/src/test/scala/ch/epfl/bluebrain/nexus/cli/SparqlClientSpec.scala]: I very much prefer this implementation over a mockito impl.

File: cli/shared/src/main/scala/ch/epfl/bluebrain/nexus/cli/types/EventEnvelope.scala (hunk 2 above)
Comment (bogdanromanx) [reply cli/shared/src/main/scala/ch/epfl/bluebrain/nexus/cli/types/EventEnvelope.scala]: I'm happy to merge this as is, but we should be able to rely on having an offset. And fail on the first event that doesn't have one. It's part of the api.

**Pull request merged**